online/src/main/scala/ai/chronon/online/MetadataStore.scala

```diff
@@ -149,88 +148,31 @@ class MetadataStore(kvStore: KVStore, val dataset: String = ChrononMetadataKey,
       },
       { gb => Metrics.Context(environment = "group_by.serving_info.fetch", groupBy = gb) })
 
-  // upload the materialized JSONs to KV store:
-  // key = <conf_type>/<team>/<conf_name> in bytes e.g joins/team/team.example_join.v1 value = materialized json string in bytes
-  def putConf(configPath: String): Future[Seq[Boolean]] = {
-    val configFile = new File(configPath)
-    assert(configFile.exists(), s"$configFile does not exist")
-    logger.info(s"Uploading Chronon configs from $configPath")
-    val fileList = listFiles(configFile)
-
-    val puts = fileList
-      .filter { file =>
-        val name = parseName(file.getPath)
-        if (name.isEmpty) logger.info(s"Skipping invalid file ${file.getPath}")
-        name.isDefined
+  def put(
+      kVPairs: Map[String, Seq[String]],
+      datasetName: String = ChrononMetadataKey,
+      batchSize: Int = CONF_BATCH_SIZE
+  ): Future[Seq[Boolean]] = {
+    val puts = kVPairs.map {
+      case (k, v) => {
+        logger.info(s"""Putting metadata for
+             |dataset: $datasetName
+             |key: $k
+             |conf: $v""".stripMargin)
+        val kBytes = k.getBytes()
+        val vBytes =
+          if (v.isEmpty) Array.emptyByteArray
+          else if (v.length == 1) v.head.getBytes()
+          else v.mkString("\n").getBytes()
```

Comment:
why do we want this? If it is already bytes, what is the use of `\n`?

Reply:
@better365 after the change the value could be a list of string. not sure what will be the value in mussel if apply getBytes on a list instead of a string. I am trying to convert the list to a string first then apply toBytes

Reply:
I see. What will be used for the list of the configs?  it will be easier to code it out here.